tests/src/Loader/binding/tracing/BinderEventListener.cs

```diff
@@ -80,34 +123,46 @@ protected override void OnEventWritten(EventWrittenEventArgs data)
             if (data.EventSource.Name != "Microsoft-Windows-DotNETRuntime")
                 return;
 
-            object GetData(string name) => data.Payload[data.PayloadNames.IndexOf(name)];
-            string GetDataString(string name) => GetData(name).ToString();
+            object GetData(string name)
+            {
+                int index = data.PayloadNames.IndexOf(name);
+                return index >= 0 ? data.Payload[index] : null;
+            };
+            string GetDataString(string name) => GetData(name) as string;
 
             switch (data.EventName)
             {
                 case "AssemblyLoadStart":
+                {
+                    var bindOperation = new BindOperation()
```

Comment:
Nit: I would also refactor this into a `Parse...` method, just like the new event - for consistency - and I think it would make the code a bit more readable.